elastic/kibana
# [Synthetics] Validation for multi url hosts/urls + more

## Summary

Addresses additional bugs in synthetics validation for lightweight monitors, including

1. Properly formatting check.request.body
2. Sending an error when multiple urls or hosts are defined
3. Properly formatting service.name
4. Properly formatting ssl.supported_protocols.

### Testing

Automated testing exists for all of these flows. To test manually via push, follow the general instructions in [this PR ](https://github.com/elastic/kibana/pull/141066) plus the following test cases

1. Send a lightweight tcp/icmp/http monitor via push with multiple urls or hosts. Observe error in the cli. Error content
```
`Multiple (hosts|urls) are not supported for (http|tcp|icmp) monitors in (stack version). Please set only 1 (host|url) per monitor. You monitor was not created or updated.
```
2. Assign a lightweight monitor `service.name`. Ensure the service name was added to the monitor. Should apply whether the key is flattend or nested
3. Send a monitor with `check.request.body`. Ensure that `syntheticsMonitor[ConfigKey.REQUEST_BODY_CHECK].value` equals the correct request body value
4. Send `ssl.supported_protocols`. This can either be an array or a comma seperated list. Ensure that whether the value is an array or a comma separated list, it is saved on the monitor as an array. 

### Checklist

Delete any items that are not applicable to this PR.
